contracts/test/ramps/wise/wiseRamp.spec.ts

```diff
@@ -0,0 +1,1914 @@
+
+          await subject();
+
+          const newIntentHash = calculateIntentHash(calculateWiseId(onRamperTwoProof.public_values.profileId), subjectDepositId, await blockchain.getCurrentTimestamp());
+          const postDeposit = await ramp.getDeposit(subjectDepositId);
+
+          expect(postDeposit.outstandingIntentAmount).to.eq(subjectAmount);
+          expect(postDeposit.remainingDeposits).to.eq(preDeposit.remainingDeposits.sub(usdc(10))); // 10 usdc difference between old and new intent
+          expect(postDeposit.intentHashes).to.include(newIntentHash);
+          expect(postDeposit.intentHashes).to.not.include(oldIntentHash);
+        });
+
+        it("should delete the original intent from the intents mapping", async () => {
+          await subject();
+
+          const intent = await ramp.intents(oldIntentHash);
+
+          expect(intent.onRamper).to.eq(ADDRESS_ZERO);
+          expect(intent.deposit).to.eq(ZERO_BYTES32);
+          expect(intent.amount).to.eq(ZERO);
+          expect(intent.intentTimestamp).to.eq(ZERO);
+        });
+
+        it("should correctly add a new intent to the intents mapping", async () => {
+          await subject();
+
+          const currentTimestamp = await blockchain.getCurrentTimestamp();
+          const intentHash = calculateIntentHash(calculateWiseId(onRamperTwoProof.public_values.profileId), subjectDepositId, currentTimestamp);
+
+          const intent = await ramp.intents(intentHash);
+
+          expect(intent.onRamper).to.eq(subjectCaller.address);
+          expect(intent.deposit).to.eq(subjectDepositId);
+          expect(intent.amount).to.eq(subjectAmount);
+          expect(intent.intentTimestamp).to.eq(currentTimestamp);
+        });
+
+        it("should update the account's current intent hash", async () => {
+          await subject();
+
+          const expectedIntentHash = calculateIntentHash(
+            calculateWiseId(onRamperTwoProof.public_values.profileId),
+            subjectDepositId,
+            await blockchain.getCurrentTimestamp()
+          );
+
+          const intentHash = await ramp.getIdCurrentIntentHash(subjectCaller.address);
+
+          expect(expectedIntentHash).to.eq(intentHash);
+        });
+
+        it("should emit an IntentPruned event", async () => {
+          await expect(subject()).to.emit(ramp, "IntentPruned").withArgs(oldIntentHash, subjectDepositId);
+        });
+
+        describe("when the reclaimable amount can't cover the new intent", async () => {
+          before(async () => {
+            timeJump = ONE_DAY_IN_SECONDS.div(2).toNumber();
+          });
+
+          after(async () => {
+            timeJump = ONE_DAY_IN_SECONDS.add(1).toNumber();
+          });
+
+          it("should revert", async () => {
+            await expect(subject()).to.be.revertedWith("Not enough liquidity");
+          });
+        });
+      });
+
+      describe("when the caller is on the depositor's denylist", async () => {
```

Comment:
Missing test cases for allow list and it's combinations with deny list

Reply:
These tests are in the WiseAccountRegistry, here just trying to show that it'll trip if `isAllowedUser` returns `false`